### PR TITLE
Clean up instrumentation folder

### PR DESCRIPTION
**Pull request by llzmb**
Changes:
- Delete obsolete files.
- Merge files.
- Fix references.
- Fix typos.
- Fix line length.
- Fix formatting.

### CODE REVIEW

**Comment by llzmb** [comment]
"README.ctx.md" and "README.ngram.md" were merged into "README.llvm.md". I already fixed all references in MD files.

There is one source code file with deprecated references left:

https://github.com/AFLplusplus/AFLplusplus/blob/0e9b2089498c2acf307bbc90ade420b33aede150/src/afl-cc.c#L1797-L1802

The correct links are (btw, I'm open to subchapter/anchor name suggestions):
- instrumentation/README.ctx.md > instrumentation/README.llvm.md#6-afl-context-sensitive-branch-coverage
- instrumentation/README.ngram.md > instrumentation/README.llvm.md#7-afl-n-gram-branch-coverage

Can I leave this to you?

**Comment by vanhauser-thc** [comment]
> Can I leave this to you?

sure, will do that once this is ready for merging :)

File: instrumentation/README.persistent_mode.md [hunk 1]
```diff
@@ -54,14 +54,16 @@ main() {
 
 }
 ```
+
 And then compile:
+
 ```
 afl-clang-fast -o fuzz_target fuzz_target.c -lwhat_you_need_for_your_target
```
Comment (llzmb) [inline instrumentation/README.persistent_mode.md:61]: Is there a space missing?

```suggestion
afl-clang-fast -o fuzz_target fuzz_target.c -l what_you_need_for_your_target
```

File: instrumentation/README.persistent_mode.md [hunk 2]
```diff
@@ -145,23 +145,24 @@ The basic structure of the program that does this would be:
...
+and going much higher increases the likelihood of hiccups without giving you any
+real performance benefits.
 
-A more detailed template is shown in `../utils/persistent_mode/.`
-Similarly to the previous mode, the feature works only with afl-clang-fast; 
-`#ifdef` guards can be used to suppress it when using other compilers.
+A more detailed template is shown in
+[utils/persistent_mode](../utils/persistent_mode). Similarly to the previous
+mode, the feature works only with afl-clang-fast; `#ifdef` guards can be used to
+suppress it when using other compilers.
```
Comment (llzmb) [inline instrumentation/README.persistent_mode.md]: What is the "previous mode"? Deferred initialization?

File: instrumentation/README.persistent_mode.md (hunk 1 above)
Comment (vanhauser-thc) [reply instrumentation/README.persistent_mode.md:61]: no :)

File: instrumentation/README.persistent_mode.md (hunk 2 above)
Comment (vanhauser-thc) [reply instrumentation/README.persistent_mode.md]: yes

**Comment by llzmb** [comment]
"README.persistent_mode.md" and "README.gcc_plugin.md" share the same two sections, almost word for word:

Deferred initialization
https://github.com/AFLplusplus/AFLplusplus/blob/dev/instrumentation/README.gcc_plugin.md#4-bonus-feature-1-deferred-initialization
https://github.com/AFLplusplus/AFLplusplus/blob/dev/instrumentation/README.persistent_mode.md#3-deferred-initialization

Persistent mode
https://github.com/AFLplusplus/AFLplusplus/blob/dev/instrumentation/README.gcc_plugin.md#5-bonus-feature-2-persistent-mode
https://github.com/AFLplusplus/AFLplusplus/blob/dev/instrumentation/README.persistent_mode.md#4-persistent-mode

Can I do one of the following:
- Link from gcc_plugin to persistent_mode?
- Link the other way round?
- Link both to a new file?
- Something else?

**Comment by vanhauser-thc** [comment]
>     * Link from gcc_plugin to persistent_mode?
this please :) thank you

File: instrumentation/README.llvm.md [hunk 3]
```diff
@@ -93,81 +100,137 @@ operating mode of AFL, e.g.:
...
+detrimental, the user can choose to augment edge coverage with just the called
+function ID, instead of the entire callstack hash.
+
+In math the coverage is collected as follows: `map[current_location_ID ^
+previous_location_ID >> 1 ^ previous_callee_ID] += 1`
+
+Set the `AFL_LLVM_INSTRUMENT=CALLER` or `AFL_LLVM_CALLER=1` environment
+variable.
+
+## 7) AFL N-Gram Branch Coverage
```
Comment (vanhauser-thc) [inline instrumentation/README.llvm.md]: AFL++ ?

File: instrumentation/README.llvm.md [hunk 4]
```diff
@@ -93,81 +100,137 @@ operating mode of AFL, e.g.:
...
+## 5) Bonus feature: 'dict2file' pass
 
-This is an early-stage mechanism, so field reports are welcome. You can send bug
-reports to <afl-users@googlegroups.com>.
+Just specify `AFL_LLVM_DICT2FILE=/absolute/path/file.txt` and during compilation
+all constant string compare parameters will be written to this file to be used
+with afl-fuzz' `-x` option.
 
-## 6) deferred initialization, persistent mode, shared memory fuzzing
+## 6) AFL Context Sensitive Branch Coverage
```
Comment (vanhauser-thc) [inline instrumentation/README.llvm.md]: AFL++

**Pull request merged**